torch/_inductor/codecache.py

```diff
@@ -158,11 +159,46 @@ def is_gcc():
     return re.search(r"(gcc|g\+\+)", cpp_compiler())
 
 
+def dry_compile_so_and_load(lib_code: str, compile_fn: Callable, vec_isa):
+    _py_load_lib = """
+import torch
+from ctypes import cdll
+cdll.LoadLibrary("__lib_path__")
+"""
+
+    key, input_path = write(lib_code, "cpp", extra="")
+    from filelock import FileLock
+
+    lock_dir = get_lock_dir()
+    lock = FileLock(os.path.join(lock_dir, key + ".lock"), timeout=LOCK_TIMEOUT)
+    with lock:
+        output_path = input_path[:-3] + "so"
+        build_fn = cpp_compile_command if compile_fn is None else compile_fn
+        build_cmd = build_fn(
+            input_path, output_path, warning_all=False, vec_isa=vec_isa
+        ).split(" ")
+        try:
+            # Check build result
+            subprocess.check_output(build_cmd, stderr=subprocess.STDOUT)
+            subprocess.check_call(
+                [
+                    "python",
```

Comment:
this wont work in cases where this was run with python3 explicitly (fbcode, osx iirc, etc) - instead, use sys.executable